pkg/yang/types_builtin.go

```diff
@@ -588,17 +696,36 @@ func (n Number) Less(m Number) bool {
 		return true
 	case n.Kind != Negative && m.Kind == Negative:
 		return false
-	case n.Kind == Negative:
-		return n.Value > m.Value
+	}
+
+	// Normalize into a common representation dictated by the largest number of
+	// fraction digits in either number. Values can be compared directly in this
+	// representation.
+	maxfd := max(n.FractionDigits, m.FractionDigits)
+	nd, md := n.toDecimal(maxfd), m.toDecimal(maxfd)
```

Comment:
I don't think there was a way to save that approach due to the overflow issue. I tried something a little different instead so toDecimal is gone.